mono/metadata/object.c

```diff
@@ -101,6 +101,50 @@ mono_runtime_object_init (MonoObject *this_obj)
 	mono_error_assert_ok (error);
 }
 
+/**
+ * mono_runtime_object_init_checked:
+ * \param this_obj the object to initialize
+ * \param error set on error.
+ * This function calls the zero-argument constructor (which must
+ * exist) for the given object and returns TRUE on success, or FALSE
+ * on error and sets \p error.
+ */
+gboolean
+mono_runtime_object_init_handle (MonoObjectHandle this_obj, MonoError *error)
+{
+	MONO_REQ_GC_UNSAFE_MODE;
+
+	HANDLE_FUNCTION_ENTER ()
+
+	MonoMethod *method = NULL;
+	MonoClass *klass = MONO_HANDLE_GETVAL (this_obj, vtable)->klass;
+
+	error_init (error);
+	method = mono_class_get_method_from_name (klass, ".ctor", 0);
+	g_assertf (method, "Could not lookup zero argument constructor for class %s", mono_type_get_full_name (klass));
+
+	gpointer raw = method->klass->valuetype
+		       ? mono_handle_unbox (this_obj)
+		       : MONO_HANDLE_RAW (this_obj);
+
+	mono_runtime_invoke_checked (method, raw, NULL, error);
+
+	HANDLE_FUNCTION_RETURN_VAL (is_ok (error));
+}
+
+#if 0 // FIXMEcoop remove or refactor
```

Comment:
Don't commit dead code, it's useless. If you think this code needs to be used, why not just use it and have it replace the existing `mono_runtime_object_init_checked `?

Reply:
Was getting nervous as to the size of the change. Originally used the refactored/smaller for but then to be more careful started copy/paste/editing the functions instead, and leaving the originals. This is either the worst option in terms of code duplication and size, or it is ok because they might get deleted.